src/main/java/bio/terra/model/DRCollectionFiles.java

```diff
@@ -85,6 +91,20 @@ private FileModel getFileModel() {
         try {
             return pathLookup("/", 1);
         } catch (ApiException ex) {
+            try {
+                // if there are no files or directories in this collection AND it is the top-level files collection,
+                // then create an empty directory object here and mark it as already enumerated instead of exiting
+                // the process with the file not found exception from the API
+                Map<String, String> errorMap = CommandUtils.getObjectMapper()
```

Comment:
I'm not understanding what this code is doing. In what case does the exception message itself contain JSON?
A more reliable way might be to extract the Response from the ApiException and look at the HTTP status code.

Reply:
I think the Swagger generated client sticks the response body in the exception message if it finds a structure, so they're the same. I see this in the ApiClient.java class, invokeApi method, around line 730.

I agree that the calling getResponseBody instead of getMessage makes clearer what I'm fetching. I've swapped the methods and also added a check on the status code for further confirmation that this is the case I'm trying to handle.

I also added more comments to walk through the logic.

Thanks for the suggestions.